src/main/java/com/salesforce/dynamodbv2/mt/util/CachingAmazonDynamoDbStreams.java

```diff
@@ -720,29 +739,26 @@ private static String toShortString(GetRecordsResult result) {
 
         // eagerly create various meters
         final String cn = CachingAmazonDynamoDbStreams.class.getSimpleName();
-        this.getRecordsTime = meterRegistry.timer(cn + ".GetRecords.Time");
-        this.getRecordsSize = meterRegistry.summary(cn + ".GetRecords.Size");
-        this.getRecordsLoadTime = meterRegistry.timer(cn + ".GetRecords.Load.Time");
-        this.getRecordsLoadSize = meterRegistry.summary(cn + ".GetRecords.Load.Size");
-        this.getRecordsLoadLockWaitTime = meterRegistry.timer(cn + ".GetRecords.Load.Lock.Time");
-        this.getRecordsLoadLockTimeouts = meterRegistry.counter(cn + ".GetRecords.Load.Lock.Timeouts");
-        this.getRecordsLoadLockTimeoutsFailures = meterRegistry.counter(cn + ".GetRecords.Load.Lock.Timeouts.Failures");
-        this.getRecordsLoadRetries = meterRegistry.summary(cn + ".GetRecords.Load.Retries");
-        this.getRecordsLoadMaxRetries = meterRegistry.counter(cn + ".GetRecords.Load.MaxRetries");
-        this.getRecordsLoadExpiredIterator = meterRegistry.counter(cn + ".GetRecords.Load.ExpiredIterator");
-        this.getRecordsUncached = meterRegistry.counter(cn + ".GetRecords.Uncached");
-        this.getShardIteratorLoadTime = meterRegistry.timer(cn + ".GetShardIterator.Load.Time");
-        this.getShardIteratorUncached = meterRegistry.counter(cn + ".GetShardIterator.Uncached");
-        GuavaCacheMetrics.monitor(meterRegistry, iteratorCache, cn + ".GetShardIterator");
-        GuavaCacheMetrics.monitor(meterRegistry, getRecordsEmptyResultCache, cn + ".EmptyResult");
-        GuavaCacheMetrics.monitor(meterRegistry, describeStreamCache, cn + ".DescribeStream");
-        GuavaCacheMetrics.monitor(meterRegistry, trimHorizonCache, cn + ".TrimHorizon");
-    }
-
-    @VisibleForTesting
-    // TODO: introduce builder/constructor variant that allows passing in cache reference
-    public Cache<String, DescribeStreamResult> getDescribeStreamCache() {
-        return describeStreamCache;
+        final String prefix = StringUtils.isNullOrEmpty(metricPrefix) ? cn : cn + "." + metricPrefix;
```

Comment:
Just curious: why did you decide to go with this option vs injecting the prefix via a `MeterRegistry` in ZOS?

Reply:
I wanted to talk to you about that, I did try implementing `MeterRegistry`, and had a couple concerns.  See below are some examples, in creating a timer, I chose to call the method that takes tags, but there are other timer methods (one that takes strings), but there wasn't an option to pass in distributionStatisticConfig for example.  Most of the registries that implement `MeterRegistry` have separate classes for each type, i.e. Composite counter.  I think we just want to add the prefix and not have custom metric types, basically wasn't sure how in depth we wanted to go with the implementation.

    @Override
    protected Timer newTimer(Id id, DistributionStatisticConfig distributionStatisticConfig,
                             PauseDetector pauseDetector) {
        return meterRegistry.timer(prefix + "." + id.getName(), id.getTags());
    }

    @Override
    protected DistributionSummary newDistributionSummary(Id id, DistributionStatisticConfig distributionStatisticConfig,
                                                         double scale) {
        return meterRegistry.summary(id.getName(), id.getTags());
    }